### ISSUE TITLE
Why is expression result reinterpreted?

### ISSUE THREAD

**Issue by michal-klimek** [HubSpot/jinjava#121]
Please consider the following code:
```
final String tpl = "The question is: {{ obj.text }}!";
final Map<String, Object> ctx = new HashMap<>();
final Map<String, Object> obj = new HashMap<>();
obj.put("text", "{{ obj.val }}");
obj.put("val", "is it a bug?");
ctx.put("obj", obj);
Jinjava jj = new Jinjava();
System.out.println(jj.render(tpl, ctx));
```
The expected output is: `The question is {{ obj.val }}!`, but `The question is: is it a bug?!` appears on my console.

I have created similar code in Jinja2 (python):
```
from jinja2 import Template
t = Template('The question is {{ obj.text }}!')
ctx = {"obj": {"text": "{{ obj.val }}", "val": "is it a bug?"}}
print t.render(ctx)
```
The render result is as expected: `The question is {{ obj.val }}!`

I would like to know if it is an intended behavior, because in my application I have some templates where expressions insert google-encoded polylines which may occasionally contain `{{  }}` characters, which in turn are interpreted as another expression with unresolvable name.

**Comment by hs-lsong** [comment]
This seems working as designed for jinjava. There is in fact a test to verify that the expression is interpreted. 
https://github.com/HubSpot/jinjava/blob/master/src/test/java/com/hubspot/jinjava/tree/ExpressionNodeTest.java#L28

It is definitely a divergence from the python version. It is unclear if anyone is depending this behavior.

**Comment by straight-shoota** [comment]
This is a potential security risk, because it allows user-generated text to be evaluated as template.

**Comment by mattcoley** [comment]
I believe this should be turned off by default. In addition to the security issue, this behavior also breaks `raw` tags. It took me a while to figure out this setting and I am guessing most developers building applications that are using jinjava will not be aware of this issue. https://github.com/HubSpot/jinjava/blob/master/src/main/java/com/hubspot/jinjava/JinjavaConfig.java#L55

**Comment by boulter** [comment]
I agree it would be safer to turn it off, but for existing templates, it could definitely break things.
